### ISSUE TITLE
False positive "should be listed in catkin_package()" for tf2_geometry_msgs

### ISSUE THREAD

**Issue by mortenfyhn** [fkie/catkin_lint#79]
The tf2_geometry_msgs package contains some library tf2/rosmsg conversion stuff, and no message definitions. If I use it, and add it as a package dependency and catkin component, I get a catkin_lint warning like this:

```
<my package>: CMakeLists.txt(16): warning: package 'tf2_geometry_msgs' should be listed in catkin_package()
     * Your package configures another package as build dependency, it
     * is listed as run_depend in your package.xml, and its name
     * suggests that it contains ROS messages. In that case, you must
     * add it to the CATKIN_DEPENDS stanza of your catkin_package()
     * You can ignore this problem with --ignore suggest_catkin_depend
catkin_lint: checked 1 packages and found 1 problems
```

I guess the problem is that a `*_msgs` package isn't guaranteed to contain messages. Maybe it's possible to check packages for message/service/action definitions instead of checking the package name?

Edit: Or is it against ROS guidelines to name a package `*_msgs` if it has no messages? In that case the linter is right and tf2_geometry_msgs should be renamed (which I suppose won't happen soon because it'll break everything).

**Comment by mortenfyhn** [comment]
Nice!
